.eslintrc.cjs

```diff
@@ -674,7 +674,7 @@ module.exports = {
     'no-this-before-super': [2],
     'no-throw-literal': [2],
     'no-undef-init': [2],
-    'no-undef': [0],
+    'no-undef': [2], // it is still needed by eslint & IDE to prompt undefined names in real time
```

Comment:
I don't think it's correct to have it enabled as per https://typescript-eslint.io/troubleshooting/faqs/eslint/#i-get-errors-from-the-no-undef-rule-about-global-variables-not-being-defined-even-though-there-are-no-typescript-errors.

Reply:
I do think it is correct to have it enabled at the moment, otherwise IDE (I am using GoLand) won't report errors, which makes difficult write correct code (for example, remove `import $`)

Reply:
> As for JetBrains IDEs: Isn't one supposed to use WebStorm or something for Typescript?

The lint result comes from `eslint` CLI.

Reply:
Or, could you show how to correctly lint the "undefined $" errors? You could just remove the `import $` from a JS file, and try to see whether there is any lint error (either from eslint or other tools)

Reply:
What do you expect when removing `import $`? The global to work, so it not raises an error?

Reply:
> What do you expect when removing `import $`? The global to work, so it not raises an error?

It should raise an error, there is no global `$`

Reply:
Right, it should as there is no global defined here and `env.jQuery` is also not set:

https://github.com/go-gitea/gitea/blob/6410c34b7f6b7ee5daf27b057b197e63fdeb9be6/.eslintrc.cjs#L51-L54

I will check later why it does not error.